app/src/main/java/space/taran/arknavigator/mvp/model/repo/kind/ResourceKind.kt

```diff
@@ -0,0 +1,79 @@
+package space.taran.arknavigator.mvp.model.repo.kind
+
+import space.taran.arknavigator.mvp.model.dao.ResourceExtra
+import space.taran.arknavigator.utils.extension
+import java.nio.file.Path
+import kotlin.reflect.full.isSubclassOf
+
+sealed class ResourceKind {
```

Comment:
Now it became a bit confusing name, but maybe it's still better than `ResourceMetaExtraImpl`...